lib/reporting/drop_off_report.rb

```diff
@@ -0,0 +1,397 @@
+          'Welcome (page viewed)',
+          'Start of proofing process',
+        ],
+        [
+          'User agreement (page viewer)',
+          'Users who clicked "Continue" on the welcome page',
+        ],
+        [
+          'Capture Document (page viewed)',
+          'Users who check the consent checkbox and click "Continue"',
+        ],
+        [
+          'Document submitted (event)',
+          'Users who upload a front and back image and click "Submit"	',
+        ],
+        [
+          'SSN (page view)',
+          'Users whose ID is authenticated by Acuant',
+        ],
+        [
+          'Verify Info (page view)',
+          'Users who enter an SSN and continue',
+        ],
+        [
+          'Verify submit (event)',
+          'Users who verify their information and submit it for Identity Verification (LN)',
+        ],
+        [
+          'Phone finder (page view)',
+          'Users who successfuly had their identities verified by LN',
+        ],
+        [
+          'Encrypt account: enter password (page view)',
+          'Users who were able to complete the physicality check using PhoneFinder',
+        ],
+        [
+          'Personal key input (page view)',
+          'Users who enter their password to encrypt their PII',
+        ],
+        [
+          'Verified (event)',
+          'Users who confirm their personal key and complete setting up their verified account',
+        ],
+      ]
+    end
+
+    def idv_started
+      data[Events::IDV_DOC_AUTH_WELCOME].count
+    end
+
+    def idv_doc_auth_welcome_submitted
+      data[Events::IDV_DOC_AUTH_WELCOME_SUBMITTED].count
+    end
+
+    def idv_doc_auth_document_captured
+      data[Events::IDV_DOC_AUTH_CAPTURED].count
+    end
+
+    def idv_doc_auth_ssn_visited
+      data[Events::IDV_DOC_AUTH_SSN_VISITED].count
+    end
+
+    def idv_doc_auth_verify_visited
+      data[Events::IDV_DOC_AUTH_VERIFY_VISITED].count
+    end
+
+    def idv_doc_auth_verify_submitted
+      data[Events::IDV_DOC_AUTH_VERIFY_SUBMITTED].count
+    end
+
+    def idv_doc_auth_phone_visited
+      data[Events::IDV_DOC_AUTH_PHONE_VISITED].count
+    end
+
+    def idv_enter_password_visited
+      data[Events::IDV_ENTER_PASSWORD_VISITED].count
+    end
+
+    def idv_enter_password_submitted
+      data[Events::IDV_ENTER_PASSWORD_SUBMITTED].count
+    end
+
+    def idv_personal_key_submitted
+      data[Events::IDV_PERSONAL_KEY_SUBMITTED].count
+    end
+
+    def as_tables
+      [
+        proofing_definition_table,
+        step_definition_table,
+        overview_table,
+        dropoff_metrics_table,
+      ]
+    end
+
+    def to_csvs
+      as_tables.map do |table|
+        CSV.generate do |csv|
+          table.each do |row|
+            csv << row
+          end
+        end
+      end
+    end
+
+    # @return [String]
+    def format_as_percent(numerator:, denominator:)
+      (100 * numerator.to_f / denominator.to_f).round(2).to_s + '%'
+    end
```

Comment:
are we planning for this spreadsheet to be emailed using our report mailer? if so, for percents we just leave them as decimals between `0.0` and `1.0` and then we can pass `format_as_percent` and it'll get pretty formatted for us

Reply:
yeah, that is the use case for this!

just to clarify, are you saying that if i pass the table into the `EmailableReport` class with decimals, and pass `float_as_percent: true,` as an option, they get pretty formatted automatically?

Reply:
Yes!
https://github.com/18F/identity-idp/blob/3b14b9d9138c82d1dc9869cd943f74beffe8dec2/app/services/reporting/account_deletion_rate_report.rb#L16-L24

Reply:
updated in https://github.com/18F/identity-idp/commit/6aaa8ec6ddfd8ec047ea564c89aa158836607f65